services/director-v2/src/simcore_service_director_v2/modules/celery.py

```diff
@@ -0,0 +1,69 @@
+import logging
+from dataclasses import dataclass
+from typing import List
+
+from celery import Celery, Task
+from celery.contrib.abortable import AbortableAsyncResult
+from fastapi import FastAPI
+from models_library.projects import ProjectID
+from models_library.settings.celery import CeleryConfig
+
+from ..models.schemas.constants import UserID
+from ..utils.client_decorators import handle_retry
+
+logger = logging.getLogger(__name__)
+
+
+def setup(app: FastAPI, settings: CeleryConfig):
+    if not settings:
+        settings = CeleryConfig.create_from_env()
+
+    def on_startup() -> None:
+        CeleryClient.create(
+            app,
+            client=Celery(
+                settings.task_name,
+                broker=settings.broker_url,
+                backend=settings.result_backend,
+            ),
+            settings=settings,
+        )
+
+    async def on_shutdown() -> None:
+        del app.state.celery_client
+
+    app.add_event_handler("startup", on_startup)
+    app.add_event_handler("shutdown", on_shutdown)
+
+
+@dataclass
+class CeleryClient:
+    client: Celery
+    settings: CeleryConfig
+
+    @classmethod
+    def create(cls, app: FastAPI, *args, **kwargs):
+        app.state.celery_client = cls(*args, **kwargs)
+        return cls.instance(app)
+
+    @classmethod
+    def instance(cls, app: FastAPI):
+        return app.state.celery_client
+
+    @handle_retry(logger)
```

Comment:
what about error handling for celery? [Doc](https://docs.celeryproject.org/en/2.4-archived/reference/celery.app.html?highlight=send_task#celery.app.App.send_task) does not mention any exception raised in ``send_task`` but would you please  double check if some of these [celery errors](https://docs.celeryproject.org/en/2.4-archived/reference/celery.exceptions.html?highlight=exceptions#module-celery.exceptions) might be an issue?

Reply:
I will add a comment. I want to double check later when doing subparts and changing how we start tasks.